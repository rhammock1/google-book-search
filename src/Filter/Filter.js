import React from 'react';

const Filter = function() {
  return (
    <form className='filter-form'>
      <fieldset>
        <legend>Filter Options</legend>
        <label htmlFor='print-type'>Print Type: </label>
        <select id='print-type'>
          <option value='all'>All</option>
          <option value='books'>Books</option>
          <option value='magazines'>Magazines</option>
        </select>
        <label htmlFor='book-type'>Book Type: </label>
        <select id='book-type'>
          <option value='partial'>Partial view</option>
          <option value='full'>Full view</option>
          <option value='free-ebooks'>Free eBooks</option>
          <option value='paid-ebooks'>Paid eBooks</option>
          <option value='ebooks'>Only eBooks</option>
        </select>
      </fieldset>
    </form>
  )
}

export default Filter;