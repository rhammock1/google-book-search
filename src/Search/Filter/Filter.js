import React from 'react';

const Filter = function() {
  return (
      <>
        
      </>)
}

export default Filter;