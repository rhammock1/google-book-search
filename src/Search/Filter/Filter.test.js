import React, { createElement } from 'react';
import ReactDOM from 'react-dom';
import Filter from './Filter';
import renderer from 'react-test-renderer';


describe('Search component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Filter />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer.create(<Filter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})