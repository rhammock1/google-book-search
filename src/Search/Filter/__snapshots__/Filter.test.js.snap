// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders the UI as expected 1`] = `
Array [
  <label
    htmlFor="prinType"
  >
    Print Type: 
  </label>,
  <select
    id="prinType"
    name="printType"
  >
    <option
      value="all"
    >
      All
    </option>
    <option
      value="books"
    >
      Books
    </option>
    <option
      value="magazines"
    >
      Magazines
    </option>
  </select>,
  <label
    htmlFor="filter"
  >
    Book Type: 
  </label>,
  <select
    id="filter"
    name="filter"
  >
    <option
      value="partial"
    >
      Partial view
    </option>
    <option
      value="full"
    >
      Full view
    </option>
    <option
      value="free-ebooks"
    >
      Free eBooks
    </option>
    <option
      value="paid-ebooks"
    >
      Paid eBooks
    </option>
    <option
      value="ebooks"
    >
      Only eBooks
    </option>
  </select>,
]
`;
